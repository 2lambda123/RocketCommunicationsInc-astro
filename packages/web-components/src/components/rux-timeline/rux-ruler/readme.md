# rux-ruler

<!-- Auto Generated Below -->


## Properties

| Property            | Attribute            | Description | Type     | Default     |
| ------------------- | -------------------- | ----------- | -------- | ----------- |
| `endDate`           | `end-date`           |             | `any`    | `undefined` |
| `interval`          | `interval`           |             | `any`    | `''`        |
| `intervalIncrement` | `interval-increment` |             | `number` | `1`         |
| `startDate`         | `start-date`         |             | `any`    | `undefined` |
| `track`             | `track`              | The track   | `string` | `'1'`       |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
