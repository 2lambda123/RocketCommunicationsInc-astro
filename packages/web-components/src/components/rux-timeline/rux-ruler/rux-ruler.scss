:host {
    display: contents;
}

.rux-track {
    display: contents;
}

.ruler-time {
    display: flex;
    align-items: center;
    padding: 8px 20px 8px;
    justify-content: center;
}

.ruler-new-day-cell {
    background-color: var(--color-background-base-default);
}

.ruler-new-day-display {
    display: flex;
    align-items: flex-end;
    background: var(--color-background-surface-header);
    text-align: left;
    padding-inline-start: 20px;

    padding-block: var(--spacing-1);

    & span {
        position: sticky;
        left: 210px;
    }
}
