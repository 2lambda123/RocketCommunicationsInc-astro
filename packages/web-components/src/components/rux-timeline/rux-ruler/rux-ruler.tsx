import { Prop, Component, Host, h } from '@stencil/core'

@Component({
    tag: 'rux-ruler',
    styleUrl: 'rux-ruler.scss',
    shadow: true,
})
export class RuxRuler {
    /**
     * The track
     */
    @Prop() track: string = '1'

    get times() {
        const times = []
        for (let i = 0; i < 24; i++) {
            times.push((i < 10 ? '0' : '') + `${i}:00`)
        }
        return times
    }

    render() {
        console.log('times', this.times)
        return (
            <Host>
                <div class="rux-ruler rux-track">
                    <span
                        class="ruler-time"
                        style={{ gridRow: `${this.track}` }}
                    ></span>
                    {this.times.map((time, index) => (
                        <span
                            class={{
                                'ruler-time': true,
                                'ruler-time__first': index === 0,
                            }}
                            style={{ gridRow: `${this.track}` }}
                        >
                            {time}
                        </span>
                    ))}
                </div>
            </Host>
        )
    }
}
