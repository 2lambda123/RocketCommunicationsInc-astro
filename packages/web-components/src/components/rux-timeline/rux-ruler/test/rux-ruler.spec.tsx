import { newSpecPage } from '@stencil/core/testing'
import { RuxRuler } from '../rux-ruler'

describe('rux-ruler', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxRuler],
            html: `<rux-ruler></rux-ruler>`,
        })

        expect(page.root).toMatchSnapshot()
    })
})
