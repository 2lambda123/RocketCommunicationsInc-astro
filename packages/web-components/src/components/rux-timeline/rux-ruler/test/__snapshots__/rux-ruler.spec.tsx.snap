// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-ruler renders 1`] = `
<rux-ruler>
  <mock:shadow-root>
    <div class="rux-ruler rux-track"></div>
  </mock:shadow-root>
</rux-ruler>
`;
